applications/compareSummary.hpp

```diff
@@ -0,0 +1,86 @@
+
+#include <ert/ecl/ecl_sum.h>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <ert/util/stringlist.h>
+#include <ert/util/int_vector.h>
+#include <ert/util/bool_vector.h>
+#include <opm/common/ErrorMacros.hpp>
+void printHelp();
+
+
+struct Deviation {
+  double relative_deviation; 
+  double absolute_deviation; 
+};
+
+
+class SummaryReader {
+private:
+  ecl_sum_type* ecl_sum1;
+  ecl_sum_type* ecl_sum2;
+  ecl_sum_type* ecl_sum_file_short;
+  ecl_sum_type * ecl_sum_file_long;
+  stringlist_type* keys1;
+  stringlist_type* keys2; 
+  stringlist_type * keys_short;
+  stringlist_type * keys_long;
+  double relative_tolerance_max;
+  double relative_tolerance_median_max;
+  std::vector<double> * referance_vec;
+  std::vector<double> * ref_data_vec;
+  std::vector<double> * checking_vec;
+  std::vector<double> * check_data_vec;
+public:
+  SummaryReader(): 
```

Comment:
This constructor seems to be implemented both in the header file and source file.